### PR TITLE
`azurerm_cosmosdb_mongo_database` doesn't call get throughput api when cosmos account is serverless

### PR DESCRIPTION
**Pull request by yupwei68**
Fix https://github.com/terraform-providers/terraform-provider-azurerm/issues/8661

=== RUN   TestAccAzureRMCosmosDbMongoDatabase_basic
=== PAUSE TestAccAzureRMCosmosDbMongoDatabase_basic
=== CONT  TestAccAzureRMCosmosDbMongoDatabase_basic
--- PASS: TestAccAzureRMCosmosDbMongoDatabase_basic (1726.68s)
=== RUN   TestAccAzureRMCosmosDbMongoDatabase_complete
=== PAUSE TestAccAzureRMCosmosDbMongoDatabase_complete
=== CONT  TestAccAzureRMCosmosDbMongoDatabase_complete
--- PASS: TestAccAzureRMCosmosDbMongoDatabase_complete (1794.50s)
=== RUN   TestAccAzureRMCosmosDbMongoDatabase_autoscale
=== PAUSE TestAccAzureRMCosmosDbMongoDatabase_autoscale
=== CONT  TestAccAzureRMCosmosDbMongoDatabase_autoscale
--- PASS: TestAccAzureRMCosmosDbMongoDatabase_autoscale (2084.84s)
=== RUN   TestAccAzureRMCosmosDbMongoDatabase_serverless
=== PAUSE TestAccAzureRMCosmosDbMongoDatabase_serverless
=== CONT  TestAccAzureRMCosmosDbMongoDatabase_serverless
--- PASS: TestAccAzureRMCosmosDbMongoDatabase_serverless (1724.47s)

### CODE REVIEW

**Review by katbyte — CHANGES_REQUESTED** [review]
Aside from one comment this LGTM 👍

File: azurerm/internal/services/cosmos/cosmosdb_mongo_database_resource.go [hunk 1]
```diff
@@ -263,3 +276,12 @@ func resourceArmCosmosDbMongoDatabaseDelete(d *schema.ResourceData, meta interfa
...
 }
+
+func cosmosdbAccountPropsIsServerless(props *documentdb.DatabaseAccountGetProperties) bool {
+	for _, v := range *props.Capabilities {
+		if *v.Name == "EnableServerless" {
+			return true
+		}
+	}
+	return false
+}
```
[the remark this answers is not in the document]
Comment (yupwei68) [reply azurerm/internal/services/cosmos/cosmosdb_mongo_database_resource.go:287]: hi kt, I've noticed there's this function. But as for this case, the parameter is `[]Capability`, `Capability` is a struct with a field Name.

[the remark this answers is not in the document]
Comment (katbyte) [reply azurerm/internal/services/cosmos/cosmosdb_mongo_database_resource.go:287]: COuld we just do this inline then as this function is not likely to be re-used elsewhere?

**Comment by yupwei68** [comment]
@katbyte Thanks for your comments. Please continue reviewing.

**Comment by jackofallops** [comment]
Hi @yupwei68 

Which region/location are you running this test in?  It fails on the test suite with:
```TestAccAzureRMCosmosDbMongoDatabase_serverless: testing.go:684: Step 0 error: errors during apply:
Error: Error creating CosmosDB Account "acctest-ca-201028162029940851" (Resource Group "acctestRG-cosmos-201028162029940851"): Error creating/updating CosmosDB Account "acctest-ca-201028162029940851" (Resource Group "acctestRG-cosmos-201028162029940851"): documentdb.DatabaseAccountsClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="BadRequest" Message="EnableServerless is not valid for Mongo accounts with ServerVersion equal to 3.2.\r\nActivityId: eb658304-a4ef-455b-821f-6dc4e56adeae, Microsoft.Azure.Documents.Common/2.11.0"```

`MongoDBv3.4` capability appears to create a `v3.2` service?

**Comment by yupwei68** [comment]
Hi @jackofallops  Previously it worked. Currently I've updated the acctest to get it passed with mongo version 3.6.
=== RUN   TestAccAzureRMCosmosDbMongoDatabase_serverless
=== PAUSE TestAccAzureRMCosmosDbMongoDatabase_serverless
=== CONT  TestAccAzureRMCosmosDbMongoDatabase_serverless
--- PASS: TestAccAzureRMCosmosDbMongoDatabase_serverless (1548.01s)

**Comment by jackofallops** [comment]
Tests passing:
image

**Review by jackofallops — APPROVED** [review]
Thanks for addressing the comments / changes @yupwei68 - LGTM now 👍

**Comment by mariussm** [comment]
Hi, the same issue exists for azurerm_cosmosdb_sql_database

**Pull request merged**